lib/puppet/defaults.rb

```diff
@@ -2288,5 +2288,14 @@ def self.initialize_default_settings!(settings)
       EOT
     }
   )
+
+  settings.define_settings(
+    :main,
+    :silence_warnings => {
+      :default  => false,
+      :type     => :boolean,
+      :desc     => "Silence warnings for puppet rersource to_yaml option",
+    }
+  )
```

Comment:
@joshcooper what are your thoughts on having an option to disable the warnings? or should we just not print the warnings by setting the `silence_warnings` parameter to `true` by default when using the `ToDataConverter.convert()` method?

Reply:
The issue with mysql module is because the provider defines a [`mysql_raw` command](https://github.com/puppetlabs/puppetlabs-mysql/blob/cb97c87ec7ad189efc5bafa32ed735119afe5e2c/lib/puppet/provider/mysql.rb#L41) and the provider's `instances` method returns the output directly](https://github.com/puppetlabs/puppetlabs-mysql/blob/cb97c87ec7ad189efc5bafa32ed735119afe5e2c/lib/puppet/provider/mysql_database/mysql.rb#L12). This used to work find when puppet's `execute` just returned the string. Commit 03ed57bd73fa44b03864ddaf24b063bd313721cf added the `ProcessOutput` subclass so as to not break existing providers. I imagine it's not uncommon for a provider to return the output as the current value for a resource. So perhaps we should check for `ProcessOutput` in https://github.com/puppetlabs/puppet/pull/8726/files#diff-99868ad9181732029dbdb330f83f89b82fc114192ddea97a8520dcd6b14d808aL113-L119 and call `output` on it. That way we don't generate a bunch of noise for this specific issue, while we *do* generate meaningful messages if a provider returns data that can't be converted to `Data`/`RichData`?